arrMethodsLab.js

```diff
@@ -0,0 +1,409 @@
+// # Exercises
+//
+// 1. Given an array of numbers, write a function that returns a new array where every element is incremented by 10.
+console.log("---------- #1 ----------")
+let arr1 = [1, 2, 3, 4, 5]
+let plusTen = []
+
+arr1.forEach((currentElem) => {
+  plusTen.push(currentElem + 10)
+}
+)
+
+console.log(plusTen)
+
+// ```
+// Input: [1,2,3,4,5]
+// Output: [11,12,13,14,15]
+// ```
+console.log(" ")
+
+// 2. Given an array, return a new array where every non-string element is converted to an empty string.
+console.log("---------- #2 ----------")
+
+let arr2 = ["a", 123, "b", "c", {name: "cat"}]
+let filteredArr = arr2.filter((currentElem) => {
+  if (typeof currentElem === "string") {
+    return currentElem
+  } else {
+    return ""
+  }
+})
+
+let filteredArr2 = []
+arr2.forEach((currentElem) => {
+  if (typeof currentElem === "string") {
+    filteredArr2.push(currentElem)
+  } else {
+    filteredArr2.push("")
+  }
+})
+
+console.log(filteredArr)
+console.log(filteredArr2)
+
+// ```
+// Input: ["a", 123, "b", "c", {name: "cat"}]
+// Output: ["a", "", "b", "c", ""]
+// ```
+console.log(" ")
+
+// 3. Write a function that returns only the even elements in an array of numbers.
+console.log("---------- #3 ----------")
+
+let arr3 = [1, 2, 3, 4, 5]
+let onlyEven = arr3.filter(currentElem => currentElem % 2 === 0)
+console.log(onlyEven)
+
+// ```
+// Input: [1,2,3,4,5]
+// Output: [2,4]
+// ```
+//
+console.log(' ')
+
+// 4. Write a function that checks if all elements in the array are the same.
+console.log("---------- #4 ----------")
+
+let arr4 = [1, 1, 1, 1, 1]
+let arr5 = [1, 1, 1, 2, 1]
+
+let sameCheck1 = arr4.every((currentElem, index) => {
+  if (currentElem[index] === currentElem[index -1]) {
+    return true
+  } else {
+    return false
+  }
+})
+
```

Comment:
Great thinking . instead of comparing to the one prior . why don't we compare every element to the first element of the array?